hardware-and-software-requirements: Update supported platforms (#5587)
pingcap/docs: hardware-and-software-requirements.md

| Linux OS Platform        | Version      |
| :-----------------------:| :----------: |
| Red Hat Enterprise Linux | 7.3 or later 7.x releases |
| CentOS                   | 7.3 or later 7.x releases |
| Oracle Enterprise Linux  | 7.3 or later 7.x releases |
| Ubuntu LTS               | 16.04 or later |

> **Note:**
>
> - For Oracle Enterprise Linux, TiDB supports the Red Hat Compatible Kernel (RHCK) and does not support the Unbreakable Enterprise Kernel provided by Oracle Enterprise Linux.
> - A large number of TiDB tests have been run on the CentOS 7.3 system, and in our community there are a lot of best practices in which TiDB is deployed on the Linux operating system. Therefore, it is recommended to deploy TiDB on CentOS 7.3 or later.
> - The support for the Linux operating systems above includes the deployment and operation in physical servers as well as in major virtualized environments like VMware, KVM and XEN.

> > - Red Hat Enterprise Linux 8.0, CentOS 8 Stream and Oracle Enterprise Linux 8.0 are not supported yet as testing of these platforms is in progress.

Reviewer: ```suggestion
> - Red Hat Enterprise Linux 8.0, CentOS 8 Stream, and Oracle Enterprise Linux 8.0 are not supported yet as the testing of these platforms is in progress.
```

> > - Support for CentOS 8 Linux is not planned as upstream support ends on 31 December 2021.

Reviewer: ```suggestion
> - Support for CentOS 8 Linux is not planned because its upstream support ends on December 31, 2021.
```

Reviewer: "as" (also "since") has multiple meanings. To avoid ambiguity in technical writing, we use "because" to indicate the cause.

> > - Support for Ubuntu 16.04 will be removed in future versions of TiDB. Upgrading to Ubuntu 18.04 of newer is strongly recommended.

Reviewer: ```suggestion
> - Support for Ubuntu 16.04 will be removed in future versions of TiDB. Upgrading to Ubuntu 18.04 or later is strongly recommended.
```

> Other Linux OS versions like Debian Linux and Fedora Linux are likely to work, but are not officially supported.

Reviewer: ```suggestion
Other Linux OS versions such as Debian Linux and Fedora Linux might work but are not officially supported.
```
